File: SimpleDept/src/main/java/org/example/simpledept/service/DeptService.java
package org.example.simpledept.service;

import lombok.RequiredArgsConstructor;
import org.example.simpledept.model.entity.basic.Dept;
import org.example.simpledept.repository.basic.DeptRepository;
import org.springframework.stereotype.Service;

import java.util.Optional;

/**
 * fileName : DeptService
 * author : GGG
 * date : 2024-04-02
 * description : 부서 서비스 클래스
 */
@Service
@RequiredArgsConstructor
public class DeptService {

    private final DeptRepository deptRepository;

    // 상세조회
    public Optional<Dept> findById(int dno) {
        Optional<Dept> optionalDept = deptRepository.findById(dno);

        return optionalDept;
    }

//    저장/수정 함수
    public void save(Dept dept) {
        deptRepository.save(dept);
    }
}

File: frontend-vue/src/services/basic/DeptService.js
// 부서 CRUD 공통 함수들 정의 하는 js
import axiosDefault from "@/utils/axiosDefaultConfig";

// 상세조회
const get = (dno) => {
  return axiosDefault.get(`/basic/dept/${dno}`);
};

// 수정함수
const update = (dno, data) => {
  return axiosDefault.put(`/basic/dept/${dno}`, data);
};

const DeptService = {
  get,
  update
};

export default DeptService;
